### PR TITLE
[Page] Consolidate se23 styles and logic

### LINKED ISSUE #9954: [Page] Consolidate se23 logic and styles

### LINKED ISSUE #10202: [Page] Remove `divider` prop
During uplift, the `divider` prop styles were overridden to [remove](https://github.com/Shopify/polaris/pull/9537) the visual spacing and divider it renders based off new designs.

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #9954, #10202.

### WHAT is this pull request doing?

Consolidates se23 beta styles and logic for `Page`.
Removes `divider` prop as the styles were modified to reset pre-uplift styles in an earlier [PR](https://github.com/Shopify/polaris/pull/9537).

### How to 🎩
[Storybook](https://5d559397bae39100201eedc1-rojuobnovi.chromatic.com/?path=/story/all-components-page--default)
[Prod Storybook](https://storybook.polaris.shopify.com/?path=/story/all-components-page--default&globals=polarisSummerEditions2023:true)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Page/Page.tsx [hunk 1]
```diff
@@ -15,17 +15,9 @@ export interface PageProps extends HeaderProps {
   fullWidth?: boolean;
   /** Decreases the maximum layout width. Intended for single-column layouts */
   narrowWidth?: boolean;
-  /** Displays a divider between the page header and the page content */
-  divider?: boolean;
```
Comment (laurkim) [inline polaris-react/src/components/Page/Page.tsx:19]: Removed this prop as the styles were modified with uplift in a previous [PR](https://github.com/Shopify/polaris/pull/9537) and the class is just overriding pre-uplift spacing back to 0.

File: polaris-react/src/components/Page/components/Header/Header.scss [hunk 2]
```diff
@@ -16,49 +16,28 @@ $action-menu-rollup-computed-width: 40px;
...
   line-height: 1;
-
-  button {
-    // stylelint-disable declaration-no-important -- generated by polaris-migrator DO NOT COPY
-    box-shadow: none !important;
-
-    &:hover,
-    &:active,
-    &:focus {
-      border: var(--p-border-width-1) solid var(--p-color-border-strong) !important;
```
Comment (laurkim) [inline polaris-react/src/components/Page/components/Header/Header.scss:54]: Removed this due to specificity issues. Verified that without this, styles for interactive states look as expected.

File: polaris.shopify.com/content/components/layout-and-structure/page.md [hunk 3]
```diff
@@ -60,9 +60,6 @@ examples:
   - fileName: page-with-content-after-title.tsx
     title: With content after title
     description: Title metadata appears immediately after the page’s title. Use it to communicate brief, important and non-interactive status information about an entire page.
-  - fileName: page-with-divider.tsx
-    title: With divider
-    description: Use when the page needs visual separation between the page header and the content.
```
Comment (laurkim) [inline polaris.shopify.com/content/components/layout-and-structure/page.md:65]: @yurm04 I had to remove this example to resolve build issues with the removed prop

File: polaris-react/src/components/Page/Page.tsx (hunk 1 above)
Comment (sophschneider) [reply polaris-react/src/components/Page/Page.tsx:19]: do you know if we are tracking prop deprecations for v12 anywhere?

**Review by sophschneider — APPROVED** [review]

File: polaris-react/src/components/Page/Page.tsx (hunk 1 above)
Comment (laurkim) [reply polaris-react/src/components/Page/Page.tsx:19]: Added `divider` prop removal to tasklist [here](https://github.com/Shopify/polaris/issues/9914) 👍

**Pull request merged**